### ISSUE TITLE
Binding to table entity without RowKey fails

### ISSUE THREAD

**Issue by azyl-scnsoft** [Azure/azure-webjobs-sdk-script#624]
When declaring input binding to table entity with RowKey as empty string, binding fails with the next error:
`Can't bind Table to type 'MyType'`

I assume this happens because of this check in TableAttributeBindingProvider in WebJobs SDK

`bool bindsToEntireTable = tableAttribute.RowKey == null;`

This forces to set RowKey even if it's not needed

**Comment by mathewc** [comment]
Can you please provide a repro function for this? It sounds like you're trying to bind an entire table to a singleton Type "MyType" which is not expected to work.

**Comment by alex-zyl** [comment]
@mathewc  Here is the repro steps:

1. Create C# function using any random template
2. Add Table Storage input binding with hardcoded PK and RK
3. Edit function.json to leave RK empty 

Expected result: 
Entity with provided PK and RK as empty string passed to function

Actual result: 
Error indexing method 'Functions.BlobTriggerCSharp1'. Microsoft.Azure.WebJobs.Host: Can't bind Table to type 'Submission#0+TypeName'.

**Comment by mathewc** [comment]
And in your method signature, you're trying to bind this `inTable` binding to a parameter that is a singleton, e.g. `MyType`? That isn't expected to work. To resolve a singleton table entity, both row key and partition key are needed. Can you please share your C# method signature?

**Comment by azyl-scnsoft** [comment]
@mathewc  

>And in your method signature, you're trying to bind this inTable binding to a parameter that is a singleton, e.g. MyType?
Yes

> To resolve a singleton table entity, both row key and partition key are needed.
Yes, but it's possible to create an entity with empty string as a row key, so there cannot be two rows that have, um, PK == "123" and RK == "".

**Comment by mathewc** [comment]
That is indeed a corner case we don't handle right now. You're saying that you've designed your table so that empty row keys are part of your design (PK with empty RK)? Why? I haven't heard of anyone doing this before.

**Comment by alex-zyl** [comment]
@mathewc  
Well, yes. We had to store one particular row with some sensitife information alongside with other rows that have both PK and RK. Using empty RK helped us to ignore this particular row when querying entities.

**Comment by mathewc** [comment]
As a workaround, I suggest that you instead specify a Filter `(PartitionKey eq '<key') and (RowKey eq '')` which we do support. Note that for this you'll need to use the new `JArray` binding available in the latest 1.0 release:

```csharp
#r "Newtonsoft.Json"

using System;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

public static void Run(string input, JArray row, TraceWriter log)
{
    var entity = row[0].ToString();
    log.Info(entity);
}
```

I've verified this works. While we could make changes to support an empty RK (those changes would have to happen in our [core SDK repo](https://github.com/Azure/azure-webjobs-sdk)), I don't think that is a priority for us right now, particularly given that I haven't heard anyone ask for this in the 3+ years the core SDK has been in existence.
